### PR TITLE
Properly stringify both large and small int64's when converting to json

### PR DESCRIPTION
**Pull request by 766C6164**
This PR contains fix (and ut's) for the following issue (https://github.com/AntelopeIO/leap/issues/579):

I expected to receive an action value is -90909090909090909. But the action return the value is -90909090909090910

  [[eosio::action]] int64_t getvalue2() {
        int64_t value = -90909090909090909;
        eosio::print("getvalue2: ", value);
        return value;
  }
You can reproduce this issue through this repo https://github.com/quocle108/return-error

### CODE REVIEW

File: libraries/libfc/src/io/json.cpp [hunk 1]
```diff
@@ -596,7 +596,7 @@ namespace fc
          {
               int64_t i = v.as_int64();
               if( format == json::output_formatting::stringify_large_ints_and_doubles &&
-                  i > 0xffffffff )
+                  (i * ((i>0)-(i<0)) > 0xffffffff))
```
Comment (heifner) [inline libraries/libfc/src/io/json.cpp:599]: I find this construct rather odd and I am a bit surprised that compilers would not give a warning about bool used in asthmatic statement.

Seems like: `(i > 0xffffffff) || (i < -0xffffffff)` would be clearer.

Comment (mikelik) [reply libraries/libfc/src/io/json.cpp:599]: how about `std::abs(i) > 0xffffffff`
probably `cmath` should be included

Comment (smlu) [reply libraries/libfc/src/io/json.cpp:599]: Casting to `uint64_t` should also do the job, i.e. `static_cast<uint64_t>(i) > 0xffffffff`

Comment (766C6164) [reply libraries/libfc/src/io/json.cpp]: Thank you, all great suggestions! Changed.

File: libraries/libfc/src/io/json.cpp [hunk 2]
```diff
@@ -595,12 +595,13 @@ namespace fc
          case variant::int64_type:
          {
               int64_t i = v.as_int64();
+              int64_t max_value(0xffffffff);
               if( format == json::output_formatting::stringify_large_ints_and_doubles &&
```
Comment (greg7mdp) [inline libraries/libfc/src/io/json.cpp:599]: I wonder if it shouldn't be `||`?

File: libraries/libfc/src/io/json.cpp [hunk 3]
```diff
@@ -595,12 +595,13 @@ namespace fc
          case variant::int64_type:
          {
               int64_t i = v.as_int64();
+              int64_t max_value(0xffffffff);
```
Comment (mikelik) [inline libraries/libfc/src/io/json.cpp:598]:
```suggestion
              constexpr int64_t max_value(0xffffffff);
```

File: libraries/libfc/src/io/json.cpp [hunk 4]
```diff
@@ -595,12 +595,13 @@ namespace fc
...
               if( format == json::output_formatting::stringify_large_ints_and_doubles &&
-                  i > 0xffffffff )
+                  (i > max_value || i < -max_value))
                  os << '"'<<v.as_string()<<'"';
               else
                  os << i;
-
+
```
[the remark this answers is not in the document]
Comment (766C6164) [reply libraries/libfc/src/io/json.cpp]: removed

File: libraries/libfc/src/io/json.cpp (hunk 3 above)
Comment (766C6164) [reply libraries/libfc/src/io/json.cpp]: added, thank you

**Review by mikelik — APPROVED** [review]

**Review by heifner — APPROVED** [review]

**Pull request merged**